lib/apiservers/portlayer/restapi/handlers/interaction_handlers.go

```diff
@@ -154,68 +139,106 @@ func (i *InteractionHandlersImpl) UnbindHandler(params interaction.InteractionUn
+		log.Errorf("%s", err.Error())
+
+		e := &models.Error{
+			Message: fmt.Sprintf("No resize connection found (id: %s): %s", params.ID, err.Error()),
+		}
+		return interaction.NewContainerResizeNotFound().WithPayload(e)
 	}
 
 	// Request a resize
 	cWidth := uint32(params.Width)
 	cHeight := uint32(params.Height)
 
-	err = connContainer.Resize(cWidth, cHeight, 0, 0)
-	if err != nil {
-		log.Errorf("InteractionHandler (%s) error: %s", params.ID, err.Error())
-		return interaction.NewContainerResizeInternalServerError()
+	if err = session.Resize(cWidth, cHeight, 0, 0); err != nil {
+		log.Errorf("%s", err.Error())
+
+		return interaction.NewContainerResizeInternalServerError().WithPayload(
+			&models.Error{Message: err.Error()},
+		)
 	}
 
 	return interaction.NewContainerResizeOK()
 }
 
+// ContainerSetStdinHandler returns the stdin
 func (i *InteractionHandlersImpl) ContainerSetStdinHandler(params interaction.ContainerSetStdinParams) middleware.Responder {
 	defer trace.End(trace.Begin(params.ID))
 
 	var ctxDeadline time.Time
 	var timeout time.Duration
 
 	// Calculate the timeout for the attach if the caller specified a deadline.  This deadline
-
 	if params.Deadline != nil {
 		ctxDeadline = time.Time(*params.Deadline)
 		timeout = ctxDeadline.Sub(time.Now())
-		log.Printf("Attempting to get ssh session for container %s stdin with deadline %s", params.ID, ctxDeadline.Format(time.UnixDate))
+		log.Debugf("Attempting to get ssh session for container %s stdin with deadline %s", params.ID, ctxDeadline.Format(time.UnixDate))
 		if timeout < 0 {
 			e := &models.Error{Message: fmt.Sprintf("Deadline for stdin already passed for container %s", params.ID)}
 			return interaction.NewContainerSetStdinInternalServerError().WithPayload(e)
 		}
 	} else {
-		log.Printf("Attempting to get ssh session for container %s stdin", params.ID)
+		log.Debugf("Attempting to get ssh session for container %s stdin", params.ID)
 		timeout = interactionTimeout
 	}
 
-	contConn, err := i.attachServer.Get(context.Background(), params.ID, timeout)
+	session, err := i.attachServer.Get(context.Background(), params.ID, timeout)
 	if err != nil {
-		err = fmt.Errorf("No stdin found (id:%s): %s", params.ID, err.Error())
 		log.Errorf("%s", err.Error())
 
-		return interaction.NewContainerSetStdinNotFound()
+		e := &models.Error{
+			Message: fmt.Sprintf("No stdin connection found (id: %s): %s", params.ID, err.Error()),
+		}
+		return interaction.NewContainerSetStdinNotFound().WithPayload(e)
 	}
 
 	detachableIn := NewFlushingReader(params.RawStream)
-	_, err = io.Copy(contConn.Stdin(), detachableIn)
+	_, err = io.Copy(session.Stdin(), detachableIn)
 	if err != nil {
-		err = fmt.Errorf("Error copying stdin (id:%s): %s", params.ID, err.Error())
 		log.Errorf("%s", err.Error())
+
+		e := &models.Error{
+			Message: fmt.Sprintf("Error copying stdin (id: %s): %s", params.ID, err.Error()),
+		}
+		return interaction.NewContainerSetStdinInternalServerError().WithPayload(e)
```

Comment:
@sflxn any idea why we weren't returning an error here? Just an oversight or an intentional choice?